relay-server/src/utils/sizes.rs

```diff
@@ -70,21 +74,25 @@ pub fn check_envelope_size_limits(config: &Config, envelope: &Envelope) -> Resul
         };
 
         if item.len() > max_size {
-            return Err(item.ty().clone());
+            return Err(if let Some(attachment_type) = item.attachment_type() {
+                attachment_type.clone().into()
+            } else {
+                item.ty().clone().into()
+            });
         }
     }
 
     if event_size > config.max_event_size() {
-        return Err(ItemType::Event);
+        return Err(ItemType::Event.into());
     }
     if attachments_size > config.max_attachments_size() {
-        return Err(ItemType::Attachment);
+        return Err(ItemType::Attachment.into());
     }
     if session_count > config.max_session_count() {
-        return Err(ItemType::Session);
+        return Err(ItemType::Session.into());
     }
     if client_reports_size > config.max_client_reports_size() {
-        return Err(ItemType::ClientReport);
+        return Err(ItemType::ClientReport.into());
```

Comment:
I think it makes more sense to use the `DiscardItemType` directly here instead of `into`.